### PR TITLE
Cosmos 0.43

### PR DESCRIPTION
**Pull request by songwongtp**
Fixed: #XXXX

bump up to cosmos 0.43
- https://github.com/cosmos/cosmos-sdk/blob/v0.43.0/CHANGELOG.md
- https://github.com/cosmos/ibc-go/blob/v1.0.0/docs/migrations/ibc-migration-043.md

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by p-rial — APPROVED** [review]

File: x/oracle/keeper/relay.go [hunk 1]
```diff
@@ -2,19 +2,22 @@ package keeper
...
 )
 
 // OnRecvPacket processes a cross chain oracle request. Data source fees
 // are collected from an escrowAddress corresponding to the given requestKey.
-func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data types.OracleRequestPacketData) (types.RequestID, error) {
+func (k Keeper) OnRecvPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	data types.OracleRequestPacketData,
+	escrowAddress sdk.AccAddress,
```
Comment (taobun) [inline x/oracle/keeper/relay.go:16]: relayer

File: x/oracle/module.go [hunk 2]
```diff
@@ -105,7 +106,7 @@ func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {}
 
 // Route returns the module's path for message route (SDK AppModule interface).
```
Comment (taobun) [inline x/oracle/module.go:107]: // Deprecated: Route returns the message routing key for the bank module.

**Review by taobun — CHANGES_REQUESTED** [review]
Also please check oracle/keys.go to remove escrow address

File: cmd/bandd/cmd/root.go [hunk 3]
```diff
@@ -71,12 +83,13 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 }
 
 func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
-	authclient.Codec = encodingConfig.Marshaler
+	cfg := sdk.GetConfig()
+	cfg.Seal()
```
Comment (taobun) [inline cmd/bandd/cmd/root.go:87]: Please check where we should seal the config? here or in `SetBech32AddressPrefixesAndBip44CoinType`

File: x/oracle/types/packets.go [hunk 4]
```diff
@@ -47,12 +44,6 @@ func (p OracleRequestPacketData) ValidateBasic() error {
 	if !p.FeeLimit.IsValid() {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, p.FeeLimit.String())
 	}
-	if strings.Contains(p.RequestKey, "/") {
-		return sdkerrors.Wrapf(ErrInvalidRequestKey, "got: %s", p.RequestKey)
-	}
-	if len(p.RequestKey) > MaxRequestKeyLength {
```
Comment (taobun) [inline x/oracle/types/packets.go:53]: Remove `MaxRequestKeyLength` constant

File: x/oracle/ibc_test.go [hunk 5]
```diff
@@ -195,19 +166,29 @@ func (suite *OracleTestSuite) TestIBCPrepareRequestNotEnoughFund() {
 		1,
 		1,
 		sdk.NewCoins(sdk.NewCoin("uband", sdk.NewInt(3000000))),
-		"beeb-request",
 		testapp.TestDefaultPrepareGas,
 		testapp.TestDefaultExecuteGas,
 	)
+
+	// Use Carol as a relayer
```
Comment (taobun) [inline x/oracle/ibc_test.go:173]: Let's send more fund to this relayer to make sure he can pay on some data sources but not enough to pay all data sources and check the balance of treasury and carol still the same.

File: x/oracle/ibc_test.go [hunk 6]
```diff
@@ -195,19 +166,29 @@ func (suite *OracleTestSuite) TestIBCPrepareRequestNotEnoughFund() {
...
+	suite.chainB.SenderAccount = suite.chainB.App.AccountKeeper.GetAccount(suite.chainB.GetContext(), carol.Address)
+
 	packet := suite.sendOracleRequestPacket(path, 1, oracleRequestPacket, timeoutHeight)
 
-	ack := channeltypes.NewErrorAcknowledgement("0uband is smaller than 1000000uband: insufficient funds")
-	err := path.RelayPacket(packet, ack.GetBytes())
+	ack := channeltypes.NewErrorAcknowledgement("10uband is smaller than 1000000uband: insufficient funds")
+	err := path.RelayPacket(packet, ack.Acknowledgement())
 	suite.Require().NoError(err) // relay committed
 }
```
Comment (taobun) [inline x/oracle/ibc_test.go:188]: Add TestIBCPrepareExceedFeeLimit too

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — COMMENTED** [review]
Can you fix `MaximumOwasmGas`?

**Review by taobun — APPROVED** [review]
LGTM

**Review by Benzbeeb — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**